From 60a5281c74b7735b957e0091dec478c185846c2a Mon Sep 17 00:00:00 2001
From: Colin Cross <ccross@android.com>
Date: Thu, 8 Mar 2012 17:57:51 -0800
Subject: [PATCH 106/634] usb: gadget: adb: do not set error flag when
 dequeuing req

When an ep_out req is dequeued because of userspace freezing,
don't set the error flag.

Change-Id: I680f1a1059b8ac2244aaa069e7d42dc44abf98e9
Signed-off-by: Colin Cross <ccross@android.com>
---
 drivers/usb/gadget/f_adb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/f_adb.c b/drivers/usb/gadget/f_adb.c
index 4433a4d..3827715 100644
--- a/drivers/usb/gadget/f_adb.c
+++ b/drivers/usb/gadget/f_adb.c
@@ -205,7 +205,7 @@ static void adb_complete_out(struct usb_ep *ep, struct usb_request *req)
 	struct adb_dev *dev = _adb_dev;
 
 	dev->rx_done = 1;
-	if (req->status != 0)
+	if (req->status != 0 && req->status != -ECONNRESET)
 		dev->error = 1;
 
 	wake_up(&dev->read_wq);
-- 
1.9.1

